### PR TITLE
[ISSUE #32870] Adding entrypoint wrapper and migrating file based and…

### PR DESCRIPTION
**Pull request by maxi297**
## What
Addresses https://github.com/airbytehq/airbyte/issues/32870

The goal was to wrap `entrypoint.launch`. However, [pytest default behavior is to capture logs](https://github.com/pytest-dev/pytest/blob/022f1b4de546c8b3529e071965555888ecf01cb4/src/_pytest/capture.py#L148) which is fine. The problem is how it capture logs : by [aggressively overwriting sys.stdout](https://github.com/pytest-dev/pytest/blob/022f1b4de546c8b3529e071965555888ecf01cb4/src/_pytest/capture.py#L394), a global variable. It clashes with the [cpython implement of log capture](https://github.com/python/cpython/blob/e08b70fab1fbc45fa498020aac522ae1d5da6136/Lib/contextlib.py#L404). Hence, there are a couple of solutions:
* Bind our entrypoint_wrapper to pytest and have the user provide `capsys` and `caplog` as arguments in every tests
* Ask testers to add `addopts = -s` in the pytest.ini
* Ask users to do `with capsys.disabled()` in each test if they use pytest
* Use `AirbyteEntrypoint.run` (which returns messages) and tell user that if the source uses `print` directly, they’ll need to update this to use the MessageRepository or re-implement a logic similar to test_scenarios does

The options have a wild variety of drawbacks. However, use `AirbyteEntrypoint.run` doesn't seem like a big lift given that we already have source.message_repository for some source. We can enable this for all (it wasn't yet because we wanted to progressively test it and we're confident enough now).

TLDR: We will only abstract the complexity for file creation and cmd argument creation. The interface between the source and the platform (std) is not an easy pick to abstract hence we will only support messages returned by `entrypoint.run` and ask devs to update their sources if they want to test those messages.

File-based and concurrent scenarios were migrated to this new entrypoint wrapper.

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/test/entrypoint_wrapper.py`
2. `airbyte-cdk/python/airbyte_cdk/entrypoint.py`
3. `airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py`
4. The rest

## 🚨 User Impact 🚨
Scenario tests were migrated and are passing so no impact there.

There is a default message repository but for sources that hadn't define one, it should always be empty and hence there should be no impact.

I added emitting message repository messages after  `source.configure` as it is often where the migration occurs and it would impact the customer if the interpreter were to break and the finally not be executed.

~~TODO: unit tests for the entrypoint_wrapper~~ Added!

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/entrypoint.py [hunk 1]
```diff
@@ -106,6 +106,9 @@ def run(self, parsed_args: argparse.Namespace) -> Iterable[str]:
                     raw_config = self.source.read_config(parsed_args.config)
                     config = self.source.configure(raw_config, temp_dir)
 
+                    yield from [
```
Comment (maxi297) [inline airbyte-cdk/python/airbyte_cdk/entrypoint.py:109]: Emitting right after `source.configure` is it's often there the config migration occurs. See example of migration [here](https://github.com/airbytehq/airbyte/pull/33104)

File: airbyte-cdk/python/unit_tests/sources/file_based/scenarios/scenario_builder.py [hunk 2]
```diff
@@ -81,9 +81,13 @@ def configured_catalog(self, sync_mode: SyncMode) -> Optional[Mapping[str, Any]]
         for stream in self.source.streams(self.config):
             catalog["streams"].append(
                 {
-                    "stream": stream.name,
```
Comment (maxi297) [inline airbyte-cdk/python/unit_tests/sources/file_based/scenarios/scenario_builder.py:84]: Format was not a valid ConfiguredAirybteCatalog

File: airbyte-cdk/python/airbyte_cdk/entrypoint.py (hunk 1 above)
Comment (girarda) [reply airbyte-cdk/python/airbyte_cdk/entrypoint.py:109]: ~are we currently not emitting those?~
this makes sense. source-s3 is just printing the message now.

**Review by girarda — APPROVED** [review]
looks good to me! 🚢

File: airbyte-cdk/python/airbyte_cdk/entrypoint.py (hunk 1 above)
Comment (maxi297) [reply airbyte-cdk/python/airbyte_cdk/entrypoint.py:109]: Just to be clear, we would emit them before emitting the command output or the first read result or if an exception is raised. However, I was afraid of the interpreter completely breaking and skipping error handling. One example of this is:

```
>>> import ast
>>> try:
...     ast.literal_eval('+0'*10**6)
... except Exception:
...     print("Exception caught")
...
zsh: segmentation fault  python
```
